design-guidelines/form-ux.md

```diff
@@ -0,0 +1,200 @@
+
+Labels may not be replaced by placeholder text unless there are enough visual cues or context clues for the user to understand what the input does. Refer to the "Label" section above for information on labeling inputs.
+
+![Input field diagram - placeholder](./img/form-ux-patterns/inputFieldDiagram-placeholder.png)
+
+### Hint content
+
+Hint text can be used to caption an input with information that helps the user fill in the correct data. Hint text can also be used to display validation errors specific to the field.
+
+Hint content and validation errors may be displayed at the same time as long as the information is not redundant. For example: hide the hint content if there is hint text that says "Enter a number", and the validation message says "This value must be a number".
+
+Hint content should be kept as short as possible.
+
+![Input field diagram - hint content](./img/form-ux-patterns/)
+
+## Input states
+
+See the [interactive component states guidelines](./interactive-element-states.md) for more infomation about the various states an input can be in.
+
+## Submitting a form
+
+Forms can be submitted using the "Enter" key, or a submit button.
+
+The form's submit button should be highly visible, and clearly labeled to communicate what will happen when the button is pressed. For example: a button that uses form data to create a new object could have a label that says "Create", and a button that uses form data to update an existing object could have a label that says "Update" or "Save".
+
+Forms that are broken into multiple parts (e.g.: tabbed forms, form with progressively disclosed fields) should only have one submit button that is visible no matter which set of fields is being displayed. For example: if a form's fields are broken up into multiple tabs, the user should have access to the "Submit" button no matter which tab is active.
+
+Once a form is submitted, the submit button should be disabled until a success or error response is returned. While the form is waiting for a response, the submit button's label may be updated to indicate the form is being processed.
+
+For highly complex configurations or forms that could have negative consequences if it's filled out incorrectly, the "Submit" action can be moved to a confirmation screen. The confirmation screen could show users a preview of the data they are about to submit and/or a warning about any undesired side-effects that may happen if the form is filled out incorrectly.
+
+After a form is successfully submitted, users should be taken back whichever screen they opened the form from. If the content on that screen is affected by the form submission, the content should either be updated or have some indication that it is in the process of being updated.
+
+## Validation
+
+A form should not be validated until a user tries to submit it. If a form cannot be submitted, validation messages should be shown that tell the user was wrong and what they can do to fix it.
```

Comment:
This is not accurate. we do field-level validation especially when we have selection-dependent fields/progressive disclosure. This should be indicated in here.